### PR TITLE
Use require and assert in smart cotracts

### PR DESCRIPTION
**Pull request by noev**
This is my first solidity PR, all feedback is appreciated.
Replacing all throws with require/assert
Implementing #604

### CODE REVIEW

File: raiden/smart_contracts/NettingChannelLibrary.sol [hunk 1]
```diff
@@ -537,9 +491,7 @@ library NettingChannelLibrary {
         returns (uint64 nonce, address recipient, bytes32 locksroot, uint256 transferred_amount)
     {
         // size of the raw message without the signature
-        if (message.length != 196) {
-            throw;
-        }
+        assert(message.length == 196);
```
Comment (LefterisJP) [inline raiden/smart_contracts/NettingChannelLibrary.sol]: This should also be a `require`. message is provided by the user/

File: raiden/smart_contracts/NettingChannelLibrary.sol [hunk 2]
```diff
@@ -505,9 +461,7 @@ library NettingChannelLibrary {
         returns (uint64 nonce, address recipient, bytes32 locksroot, uint256 transferred_amount)
     {
         // size of the raw message without the signature
-        if (message.length != 268) {
-            throw;
-        }
+        assert(message.length == 268);
```
Comment (LefterisJP) [inline raiden/smart_contracts/NettingChannelLibrary.sol]: This should also be a `require`. message is provided by the user/

File: raiden/smart_contracts/NettingChannelLibrary.sol [hunk 3]
```diff
@@ -381,15 +343,11 @@ library NettingChannelLibrary {
...
-                throw;
-            }
+            require(self.token.transfer(counter_party.node_address, counter_amount));
         }
 
         if (closer_amount > 0) {
-            if (!self.token.transfer(closing_party.node_address, closer_amount)) {
-                throw;
-            }
+            require(self.token.transfer(closing_party.node_address, closer_amount));
```
Comment (LefterisJP) [inline raiden/smart_contracts/NettingChannelLibrary.sol]: Don't wrap those in `require`. They are external calls and as such the `if` check is fine.

File: raiden/smart_contracts/NettingChannelLibrary.sol [hunk 4]
```diff
@@ -381,15 +343,11 @@ library NettingChannelLibrary {
         closer_amount = total_deposit - counter_amount;
 
         if (counter_amount > 0) {
-            if (!self.token.transfer(counter_party.node_address, counter_amount)) {
-                throw;
-            }
+            require(self.token.transfer(counter_party.node_address, counter_amount));
```
Comment (LefterisJP) [inline raiden/smart_contracts/NettingChannelLibrary.sol]: Don't wrap those in `require`. They are external calls and as such the `if` check is fine.

File: raiden/smart_contracts/NettingChannelLibrary.sol [hunk 5]
```diff
@@ -43,26 +43,22 @@ library NettingChannelLibrary {
 
 
     modifier notSettledButClosed(Data storage self) {
-        if (self.settled > 0 || self.closed == 0)
-            throw;
+        assert(self.settled <= 0 && self.closed > 0);
```
Comment (LefterisJP) [inline raiden/smart_contracts/NettingChannelLibrary.sol]: Should be a `require`. All these are states of the contract which can be enountered, not invalid states that can never be reached.

File: raiden/smart_contracts/NettingChannelLibrary.sol [hunk 6]
```diff
@@ -43,26 +43,22 @@ library NettingChannelLibrary {
...
         _;
     }
 
     modifier stillTimeout(Data storage self) {
-        if (self.closed + self.settle_timeout < block.number)
-            throw;
+        assert(self.closed + self.settle_timeout >= block.number);
```
Comment (LefterisJP) [inline raiden/smart_contracts/NettingChannelLibrary.sol]: Should be a `require`. All these are states of the contract which can be enountered, not invalid states that can never be reached.

File: raiden/smart_contracts/NettingChannelLibrary.sol [hunk 7]
```diff
@@ -43,26 +43,22 @@ library NettingChannelLibrary {
...
         _;
     }
 
     modifier timeoutOver(Data storage self) {
-        if (self.closed + self.settle_timeout > block.number)
-            throw;
+        assert(self.closed + self.settle_timeout <= block.number);
```
Comment (LefterisJP) [inline raiden/smart_contracts/NettingChannelLibrary.sol]: Should be a `require`. All these are states of the contract which can be enountered, not invalid states that can never be reached.

File: raiden/smart_contracts/NettingChannelLibrary.sol [hunk 8]
```diff
@@ -43,26 +43,22 @@ library NettingChannelLibrary {
...
         _;
     }
 
     modifier channelSettled(Data storage self) {
-        if (self.settled == 0)
-            throw;
+        assert(self.settled != 0);
```
Comment (LefterisJP) [inline raiden/smart_contracts/NettingChannelLibrary.sol]: Should be a `require`. All these are states of the contract which can be enountered, not invalid states that can never be reached.

**Comment by LefterisJP** [comment]
Hello @noev! Thank you for your interest in Raiden. Please check my comments in your PR.

Also can you please rebase on top of latest master instead of merging master into your branch.

**Comment by noev** [comment]
Thanks for the feedback. It should be fixed now, what do you think?

**Comment by LefterisJP** [comment]
Looks good to me now. I rebased on top of latest master for you.

**Pull request merged**